gamemode/modules/f1menu/cl_searchbox.lua

```diff
@@ -6,15 +6,18 @@ function PANEL:Init()
     self.BaseClass.Init(self)
     self.F1Down = true
     self:SetFont("DarkRPHUD2")
-    self:SetTextColor(Color(255,255,255,255))
-    self:SetCursorColor(Color(255,255,255,255))
 
     self.lblSearch = vgui.Create("DLabel", self)
     self.lblSearch:SetFont("DarkRPHUD2")
-    self.lblSearch:SetColor(Color(200, 200, 200, 200))
     self.lblSearch:SetText(DarkRP.getPhrase("f1Search"))
     self.lblSearch:SizeToContents()
     self.lblSearch:SetPos(5)
+
+    if GAMEMODE.Config.DarkRPSkin == "DarkRP" then
+        self:SetTextColor(Color(255,255,255,255))
+        self:SetCursorColor(Color(255,255,255,255))
+        self.lblSearch:SetColor(Color(200, 200, 200, 200))
+    end
```

Comment:
> That should never happen. If it did then existing DarkRP SKIN fields would also have the issue and I haven't seen a complaint about it. If you found a way to reproduce this bug however, then it would definitely be worth reporting it.

I know it shouldn't, but it seems to in some cases. I believe it's to do with [ForceDermaSkin](http://wiki.garrysmod.com/page/GM/ForceDermaSkin) being used, but the affected panels using [SetSkin](http://wiki.garrysmod.com/page/Panel/SetSkin), to something else. I can't confirm though, not looked into it properly.

> Addons, future DarkRP updates etc. could change this.

In the case of addons, that's their decision, and the Server Operator's too if they wish for that. In the case of future DarkRP updates, that's not for me to decide. My intention here was to keep changes to a minimum whilst providing a little more flexibility for others wishing to use their own skin, without having to compromise too much due to hard coding.

> The current transparent background for text entries currently is applied to all DTextEntry elements with the DarkRP skin which is why a global colour change makes sense.

Fair enough, that makes sense. My knowledge on such within DarkRP is very limited, especially compared with yours. If you'd like, I can revert my change to the F1 menu so you can instead make a pull with yours instead.

> The only case I see it being a problem, because of the transparent background of the element, is if someone used a light background in the parent but then they are no longer using the DarkRP colour scheme (ie. they are using something like a hardcoded SetBackgroundColor/SetBGColor)

I don't quite understand. Are you referring to people who actually modify how the panel looks via modifying the F1 menu itself, such as making it a module of theirs?